aggregator_core/src/datastore.rs

```diff
@@ -3717,7 +3713,7 @@ impl<C: Clock> Transaction<'_, C> {
                 WHERE task_id = (SELECT id FROM tasks WHERE task_id = $1) AND client_timestamp < $2
                 AND NOT EXISTS(
                     SELECT FROM report_aggregations
-                    WHERE report_aggregations.client_report_id = client_reports.id)",
+                    WHERE report_aggregations.client_report_id = client_reports.report_id)  -- this join is very slow, fix before deploying",
```

Comment:
The first instance is test-only code that (TBH) should not have been committed yet for YAGNI reasons, so I don't mind losing performance on it. Of course, instead of maintaining this code poorly, I should instead just remove it -- I sent #1508 for this.

The second one is in GC code that will be replaced by further work in #1467. I added a reference to the issue in the TODO.